### ISSUE TITLE
15.2.0 - Lanes has an issue where using join causes a hard crash

### ISSUE THREAD

**Issue by rayaman** [rayaman/multi#30]

**Comment by rayaman** [comment]
Issue fixed by using a pcall around the join call. Now it doesn't even error... How about that

**Comment by rayaman** [comment]
Not actually fixed

**Comment by rayaman** [comment]
Seems like the issue is when the thread itself is cleaning up. Not the join call. Shortly after the thread ends Lua crashes. Not sure what is causing this. Will have to do more tests before filing an issue on the lualanes repo

**Comment by rayaman** [comment]
This isn't a good solution, but I decided to just make it so threads created using the multi library simply never end. Since the cancelation/termination of a lane seems to cause this error sometimes. 

The sometimes is what gets me. What is going on with my system? Is this a configuration of my windows system that is causing this? Does this error happen on other OS's? So much testing has to be done.

**Comment by rayaman** [comment]
So after some testing, the other lane may be touching data in the main thread causing issues

**Comment by rayaman** [comment]
This is probably not a lanes issue. When isolated lanes are created they terminate without issues. I'm not sure what is causing this to act like this. More testing needs to be done.

**Comment by rayaman** [comment]
So it might have been an issue with my drive. After changing drive labels I was unable to reproduce this issue. And I tried, I got so mad that I couldn't recreate the issue with the same code and the only thing that changed on my system is me changing my drive label since I was having problems deleting some files that were always in use for some reason. Now my code won't crash even though it has been completely reverted. Some of the optimizations I made in the process of trying to fix this will stay, but it's "Working now"
